### ISSUE TITLE
one question about Dynamic Point Fields

### ISSUE THREAD

**Issue by 2019EPWL** [sergeyprokudin/sergeyprokudin.github.io#1]
Hi, thank you for your excellent work, I would like to ask whether d in equation (13) of the AIAP constraint is based on Euclidean distance or geodesic distance.

**Comment by sergeyprokudin** [comment]
Hi @2019EPWL, thanks for your interest! It's based on the Euclidean distance.

**Comment by 2019EPWL** [comment]
thanks for your reply, I would like to ask about the difference between the AIAP used in your work and ARAP used in the baseline NDP

**Comment by sergeyprokudin** [comment]
Hi @2019EPWL, I assume you are referring to the deformability regularization term of NDP, defined by Eq. 7 in the [paper](https://arxiv.org/pdf/2205.12796.pdf). It is based on some confidence scores of the success of the deformation predicted by the network and is quite different from our AIAP formulation, which is defined simply using canonical and deformed clouds. Here's the implementation of our AIAP regularisation:
```
import torch 
from pytorch3d.ops.knn import knn_points
from pytorch3d.loss.chamfer import chamfer_distance

l1_loss = torch.nn.L1Loss()

def isometric_loss(x_canonical, x_deformed, n_neighbors=5):

  _, nn_ix, _ = knn_points(x_canonical.unsqueeze(0), 
                           x_canonical.unsqueeze(0), 
                           K=n_neighbors, 
                           return_sorted=True)

  dists_canonical = torch.cdist(x_canonical[nn_ix], x_canonical[nn_ix])
  dists_deformed = torch.cdist(x_deformed[nn_ix], x_deformed[nn_ix])

  loss = l1_loss(dists_canonical, dists_deformed)
  
  return loss
```

You may also be referring to [the ARAP cost](https://github.com/rabbityl/DeformationPyramid/blob/4e160792e5b7df8c3a78ca6b24efacdb14c33a97/model/loss.py#L227), which is implemented as a part of [embedded deformation baseline](https://people.inf.ethz.ch/~sumnerb/research/embdef/Sumner2007EDF.pdf) in the same NDP repository. AIAP and ARAP are related in the sense that they try to minimize the distortion of the mesh during the deformation, but the exact formulations of constraints are quite different, and lead to different final deformations. E.g., one apparent difference is that we do not need to necessarily parametrise our deformation as SE3 (R_i, t_i) for the isometric loss case. 

Hope this helps, and please let me know if you have additional questions.

-Sergey

**Comment by 2019EPWL** [comment]
I got it, thank you very much!
